.github/workflows/ci-tests.yml

```diff
@@ -85,13 +81,29 @@ jobs:
           -jar jobs-api-server/build/libs/jobs-api-server-*-boot.jar > ${{ env.JOBS_LOG_DIR }} &
 
       - name: Validate APIML setup
+        # poll for 3 minutes, if GW not healthy after 2 minutes then exit
         run: |
-          sleep 2m # let APIML finish setting up
-          gwHealth=$(curl -k -sS https://localhost:${{ env.GATEWAY_PORT }}/application/health)
-          echo $gwHealth
-          gatewayUp=$(echo $gwHealth | awk -F{\"status\":\" '{print $2}' | awk -F\" '{print $1}')
-          if [ "$gatewayUp" != "UP" ]; then
-            echo ">>>>>Gateway is not up"
+          set +e # curl -s doesn't swallow the error alone
+          attemptCounter=1
+          maxAttempts=18
+          until [ $attemptCounter -gt $maxAttempts ]; do
+            sleep 10
+            gwHealth=$(curl -k -s https://localhost:${{ env.GATEWAY_PORT }}/application/health)
+            echo "Polling for GW health: $attemptCounter"
+            echo $gwHealth
+
+            gatewayUp=$(echo $gwHealth | awk -F{\"status\":\" '{print $2}' | awk -F\" '{print $1}')
```

Comment:
the ubuntu runner should have jq installed which is more reliable to handle json comparing to awk. https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-Readme.md

also `echo "${gwHealth}"` is more reliable to handle string especially if the string has spaces/tabs/line feeds, etc.

So this should work better:
```
gatewayUp=$(echo "${gwHealth}" | jq -r '.status')
```